sound/soc/sof/sof-audio.c

```diff
@@ -490,10 +500,47 @@ int sof_machine_register(struct snd_sof_dev *sdev, void *pdata)
 	mach = plat_data->machine;
 	size = sizeof(*plat_data->machine);
 
-	/* register machine driver, pass machine info as pdata */
-	plat_data->pdev_mach =
-		platform_device_register_data(sdev->dev, drv_name,
-					      PLATFORM_DEVID_NONE, mach, size);
+	/* register machine driver */
```

Comment:
I know it's the same way as before but technically we register the machine device here.

Reply:
ack